show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgismaps/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,361 @@
+                val floor = (locationProperties["floor"] ?: "").toString()
+                val positionSource = (locationProperties["positionSource"] ?: "").toString()
+                val transmitterCount = (locationProperties["transmitterCount"] ?: "").toString()
+                val satelliteCount = (locationProperties["satelliteCount"] ?: "").toString()
+
+                // check if current floor hasn't been set or if the floor has changed
+                val newFloor = floor.toInt()
+                if (currentFloor == null || currentFloor != newFloor) {
+                    currentFloor = newFloor
+                    // update layer's definition express with the current floor
+                    val layerList = mapView.map?.operationalLayers
+                    layerList?.forEach { layer ->
+                        val name = layer.name
+                        if (layer is FeatureLayer && (name == "Details" || name == "Units" || name == "Levels")) {
+                            layer.definitionExpression = "VERTICAL_ORDER = $currentFloor"
+                        }
+                    }
+                }
+                // set up the message with floor properties to be displayed to the textView
+                var locationPropertiesMessage =
+                    "Floor: $floor, Position-source: $positionSource, " +
+                            "Horizontal-accuracy: " + location.let {
+                        DecimalFormat(".##").format(
+                            it.horizontalAccuracy
+                        )
+                    } + "m, "
+                if (positionSource == Location.SourceProperties.Values.POSITION_SOURCE_GNSS) {
+                    locationPropertiesMessage += "Satellite-count: $satelliteCount"
+                } else if (positionSource == "BLE") {
+                    locationPropertiesMessage += "Transmitter-count: $transmitterCount"
+                }
+                textView.text = locationPropertiesMessage
+            }
+        }
+
+        lifecycleScope.launch {
+            // Handle status changes of IndoorsLocationDataSource
+            locationDisplay.dataSource.status.collect { status ->
+                when (status) {
+                    LocationDataSourceStatus.Starting -> progressBar.visibility = View.VISIBLE
+                    LocationDataSourceStatus.Started -> progressBar.visibility = View.GONE
+                    LocationDataSourceStatus.FailedToStart -> {
+                        progressBar.visibility = View.GONE
+                        showError("Failed to start IndoorsLocationDataSource")
+                    }
+                    LocationDataSourceStatus.Stopped -> {
+                        progressBar.visibility = View.GONE
+                        mapView.locationDisplay.dataSource.stop()
+                        showError("IndoorsLocationDataSource stopped due to an internal error")
+                    }
+                    else -> null
```

Comment:
```suggestion
                    else -> {}
```

I think it's okay to have a empty block instead of null to avoid the warning.